txscript/sign.go

```diff
@@ -14,6 +14,57 @@ import (
 	"github.com/btcsuite/btcutil"
 )
 
+// RawTxInWitnessSignature returns the serialized ECDA signature for the input
+// idx of the given transaction, with the hashType appended to it. This
+// function is identical to RawTxInSignature, however the signature generated
+// signs a new sighash digest defined in BIP0143.
+func RawTxInWitnessSignature(tx *wire.MsgTx, sigHashes *TxSigHashes, idx int,
+	amt int64, subScript []byte, hashType SigHashType,
+	key *btcec.PrivateKey) ([]byte, error) {
+
+	parsedScript, err := parseScript(subScript)
+	if err != nil {
+		return nil, fmt.Errorf("cannot parse output script: %v", err)
+	}
+
+	hash := calcWitnessSignatureHash(parsedScript, sigHashes, hashType, tx,
+		idx, amt)
+	signature, err := key.Sign(hash)
+	if err != nil {
+		return nil, fmt.Errorf("cannot sign tx input: %s", err)
+	}
+
+	return append(signature.Serialize(), byte(hashType)), nil
+}
+
+// WitnessSignatureScript creates an input witness stack for tx to spend BTC
+// sent from a previous output to the owner of privKey using the p2wkh script
+// template. The passed transaction must contain all the inputs and outputs as
+// dictated by the passed hashType. The signature generated observes the new
+// transaction digest algorithm defined within BIP0143.
+func WitnessScript(tx *wire.MsgTx, sigHashes *TxSigHashes, idx int, amt int64,
```

Comment:
Function name doesn't match comment.  However, I would probably call it `WitnessSignature` since it is a stack and there are other functions named according to the scheme `FooScript` which actually return scripts versus a witness stack.  I suspect it should also return `wire.TxWitness` since that's what it is.